packages/builders/src/interactions/slashCommands/mixins/ApplicationCommandOptionWithChoicesMixin.ts

```diff
@@ -1,63 +1,28 @@
-import { s } from '@sapphire/shapeshift';
-import { ApplicationCommandOptionType, type APIApplicationCommandOptionChoice } from 'discord-api-types/v10';
+import type { APIApplicationCommandOptionChoice } from 'discord-api-types/v10';
 import { normalizeArray, type RestOrArray } from '../../../util/normalizeArray.js';
-import { localizationMapPredicate, validateChoicesLength } from '../Assertions.js';
 
-const stringPredicate = s.string().lengthGreaterThanOrEqual(1).lengthLessThanOrEqual(100);
-const numberPredicate = s.number().greaterThan(Number.NEGATIVE_INFINITY).lessThan(Number.POSITIVE_INFINITY);
-const choicesPredicate = s
-	.object({
-		name: stringPredicate,
-		name_localizations: localizationMapPredicate,
-		value: s.union([stringPredicate, numberPredicate]),
-	})
-	.array();
+export interface ApplicationCommandOptionWithChoicesData {
+	choices?: APIApplicationCommandOptionChoice<number | string>[];
+}
 
 /**
  * This mixin holds choices and autocomplete symbols used for options.
  */
-export class ApplicationCommandOptionWithChoicesMixin<ChoiceType extends number | string> {
-	/**
-	 * The choices of this option.
-	 */
-	public readonly choices?: APIApplicationCommandOptionChoice<ChoiceType>[];
-
-	/**
-	 * The type of this option.
-	 *
-	 * @privateRemarks Since this is present and this is a mixin, this is needed.
-	 */
-	public readonly type!: ApplicationCommandOptionType;
+export class ApplicationCommandOptionWithChoicesMixin {
```

Comment:
afaik TypeScript allows us (since a rather recent version) to export a value with a specific generic type:

```ts
declare function foo<Bar>(value: Bar): void;

const fn = foo<string>;
//    ^?: (value: string) => void
```

Can't we use this feature in the mixins so we can do `ApplicationCommandOptionWithChoicesMixin<number>` or `ApplicationCommandOptionWithChoicesMixin<string>`, that way we won't need to extend the methods, which we may forget in the future to properly implement?

Reply:
that sounds interesting, I'll have a look

Reply:
So, you are right, TIL, but I just realized the actual issue. The old `@mixin(A, B)` approach with interface merging is not trivial to accomplish with the new `data` field approach, which is why I swapped to `extends Mixin(A, B)`, which does not work with generics at all, so we're out of luck here, unfortunately, unless you have a better idea.

Reply:
Found a way to do this 🎉